smartsim/entity/model.py

```diff
@@ -46,6 +46,24 @@
 logger = get_logger(__name__)
 
 
+def _read_model_parameters(params_dict: t.Dict[str, t.Any]) -> t.Dict[str, str]:
+    """Convert the values in a params dict to strings
+    :raises TypeError: if params are of the wrong type
+    :return: param dictionary with values and keys cast as strings
+    """
+    param_names: t.List[str] = []
+    parameters: t.List[t.List[str]] = []
+    for name, val in params_dict.items():
+        param_names.append(name)
+        if isinstance(val, (int, str)):
```

Comment:
All number types should be valid, so let's just abstract it to:
```
import numbers
...
if isinstance(val, (str, numbers.Number)):
```

Alternatively, if we wanted to be even more generic we could just check to see if the object can be converted to a string:

```
if hasattr(val, "__str__"):
...
else:
  raise TypeError(f"Parameter {name} cannot be converted to a string")
```

Reply:
changed to numbers.number